Library/ViewModels/PledgePaymentMethodsViewModel.swift

```diff
@@ -519,45 +512,53 @@ private func pledgePaymentSheetMethodCellDataAndSelectedCardSetupIntent(
-  let preexistingPaymentSheetCardDataUnselected: [PaymentSheetPaymentMethodCellData] = {
-    var updatedPaymentSheetPaymentMethodData = paymentMethodData.paymentSheetPaymentMethodsCellData
+  let preexistingPaymentSheetCardDataUnselected: [PledgePaymentMethodCellData] = {
+    var updatedPaymentSheetPaymentMethodData = paymentMethodData.newPaymentMethods
 
     updatedPaymentSheetPaymentMethodData = updatedPaymentSheetPaymentMethodData.map { data in
       (
-        image: data.image,
-        redactedCardNumber: data.redactedCardNumber,
-        clientSecret: data.clientSecret,
+        card: data.card,
+        isEnabled: data.isEnabled,
         isSelected: false,
-        isEnabled: true
+        projectCountry: data.projectCountry,
+        isErroredPaymentMethod: data.isErroredPaymentMethod
       )
     }
 
     return updatedPaymentSheetPaymentMethodData
   }()
 
   let updatedDataWithSelection: PledgePaymentMethodsAndSelectionData = {
-    guard let newestPaymentSheetPaymentMethod = paymentMethodData.paymentSheetPaymentMethodsCellData.first
+    guard let newestPaymentSheetPaymentMethod = paymentMethodData.newPaymentMethods.first
     else {
       return paymentMethodData
     }
 
     // We know the first new payment sheet card added is the newest one, so by default select it.
     var data = preexistingPaymentSheetCardDataUnselected
 
-    let updatedSelectedPaymentSheetPaymentMethod = data[0]
-      |> \.isSelected .~ true
+    var updatedSelectedPaymentSheetPaymentMethod = data[0]
 
-    data[0] = updatedSelectedPaymentSheetPaymentMethod
+    if updatedSelectedPaymentSheetPaymentMethod.isEnabled {
+      updatedSelectedPaymentSheetPaymentMethod.isSelected = true
+    }
 
-    let usePaymentIntent = paymentSheetType == .paymentIntent
+    data[0] = updatedSelectedPaymentSheetPaymentMethod
 
     var updatePaymentMethodData = paymentMethodData
-    updatePaymentMethodData.paymentMethodsCellData = preexistingCardDataUnselected
-    updatePaymentMethodData.paymentSheetPaymentMethodsCellData = data
-    updatePaymentMethodData
-      .selectedPaymentMethod = usePaymentIntent ?
-      .paymentIntentClientSecret(newestPaymentSheetPaymentMethod.clientSecret) :
-      .setupIntentClientSecret(newestPaymentSheetPaymentMethod.clientSecret)
+    updatePaymentMethodData.existingPaymentMethods = preexistingCardDataUnselected
+    updatePaymentMethodData.newPaymentMethods = data
+
+    if newestPaymentSheetPaymentMethod.isEnabled {
+      if let stripeCardId = newestPaymentSheetPaymentMethod.card.stripeCardId {
+        updatePaymentMethodData
+          .selectedPaymentMethod = .savedCreditCard(newestPaymentSheetPaymentMethod.card.id, stripeCardId)
+      } else {
+        assert(false, "Expected stripeCardId to be set. Late pledges may fail if this value is missing.")
+        updatePaymentMethodData
+          .selectedPaymentMethod = .savedCreditCard(newestPaymentSheetPaymentMethod.card.id, "")
```

Comment:
My goal was to make sure if `stripeCardId` failed anywhere, the code would still _mostly_ work, since  `stripeCardId` is only important for late pledges.

Reply:
I like that! nit: I do think it'd be nice to pull this out into a helper function, though, considering we're doing basically the same checks in about 5 places.

Reply:
Good suggestion, fixed!